torch/_inductor/codecache.py

```diff
@@ -1945,6 +1961,60 @@ def convert_arg(arg: Any) -> Any:
         return torch._C._aoti.unsafe_alloc_void_ptr_from_tensor(result)
 
 
+# Precompiled headers are persistent past program runtime, but associated with one
+# specific compiler version and set of flags.  We explicitly use default_cache_dir here
+# because these headers need to be global, rather than ignored by fresh_inductor_cache.
+_HEADER_DIR = os.path.join(default_cache_dir(), "precompiled_headers")
+_COMPILED_HEADERS: OrderedSet[str] = OrderedSet()
+
+
+def _precompile_header(
+    header: str, hashable_cmd_line: str, compiler: str, compile_command: dict[str, Any]
+) -> str:
+    compiler_version = get_compiler_version_info(compiler)
+    header_hash, header_full_path = write(
+        content=f"#include <{header}>\n",
```

Comment:
I think it makes sense to hash on the actual header file content (as a transitive closure)?

Reply:
@desertfire I considered this, but part of the issues is that the contents of headers further down the stack could change without affecting this top-level header, and we wouldn't appropriately rebuild in that case. I spent some time listing out scenarios that would invalidate a precompiled header, and I'm reasonably confident that all of them are covered by including the extra hashable content below.